### ISSUE TITLE
[Surge of Light] Being used when it shouldn't be.

### ISSUE THREAD

**Issue by Keidarn** [dalaranwow/dalaran-wow#5280]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
When Renew procs Surge of Light in the middle of a Flash Heal cast, Surge of Light is lost.
**Expected behaviour**: Tell us what should happen instead
[Surge of Light] should be used on the next casting of Flash of Light, not on the next heal of Flash of Light
**Steps to reproduce the problem**: Assuming best case scenario,

1. Cast Renew on a target with the Surge of Light talent.
2. Cast Flash Heal as Renew gives out its health.
3. When Renew crits its HoT, Surge of Light procs
4. As Flash Heal finishes its cast, Surge of Light registers as being used and is taken from your current buffs.

**Include proofs for this behaviour**
I do not have the ability to provide proof at this time. It requires video capture software, which I do not own.

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Byssarod** [comment]
First Keidarn, I want correct you, Renew ticks can not be critical heal. Only instant from Empowered renew in holy tree which is initial so you cant reproduce issue like that. However the issue is correct. Surge of Light buff is consumed if it proc while casting Flash Heal. To reproduce problem you have to get critical heal from Prayer of Mending while you cast Flash Heal, which proc Surge of Light. When cast is finished the Surge of Light is consumed by casted Flash Heal.

**Comment by kinpie** [comment]
Byssarod is correct, 
it's rather unfortunate that your free SoL gets eaten by your cast.
